### ISSUE TITLE
automate the screenshots generation for l10n QA process

### ISSUE THREAD

**Issue by chianger** [mozilla-tw/FirefoxLite#1940]

**Comment by weslyhuang** [comment]
a good to have when we want to add more locales

**Comment by josephyu1110** [comment]
Compare [Fastlane](https://docs.fastlane.tools/getting-started/android/screenshots/) and [Espresso Screenshot](https://developer.android.com/reference/android/support/test/runner/screenshot/Screenshot) to determine which is better for us.
[Result](https://docs.google.com/spreadsheets/d/1bHmz0eZBrrzTLa1LzSgjXR_I4fYJ8q9FtNjtdfXmkSI/edit#gid=728151932)

**Comment by josephyu1110** [comment]
[2.4 QA manual screenshot list.](https://drive.google.com/drive/folders/1JFmCYBME2G-Q5kl8i5ZeAlh_fdopLaYz)
[2.4 Auto screenshot file mapping.](https://docs.google.com/spreadsheets/d/1bHmz0eZBrrzTLa1LzSgjXR_I4fYJ8q9FtNjtdfXmkSI/edit#gid=618284866)

**Comment by josephyu1110** [comment]
[Fastlane and screengrab setup note.](https://docs.google.com/document/d/1M6R9QEaiYZOKhy_uhzY5j5ilEUjz55aaut1cNIjzjKw/edit#heading=h.9oj7nblm5kah)

**Comment by cnevinc** [comment]
TLDR: we'll merge the PR #2285 and run screengrab on the local machine on demand.

After doing some survey, we want to stop here. We've spent too much time.
We are almost there, but screengrab needs to be run using fastlane command: "fastlane screengrab"
Although we can hack around the emulator startup(vritual device task can only run Gradle command), install extra apk(reinstall_app  setting is required for switching locale), and disable animaitons in the emuator (using Scripts task).... we decided to stop here. After all we can still get the screenshots locally.

Here's the last work flow we've tested: https://app.bitrise.io/build/34fa9a81080fc87f
If someday we have time, we'll pick the work up and continue.
For the reference, below is the bitrise.yml (need to set env-var "auto_screenshot " to "1")

```
primary_screenshot:
    steps:
    - activate-ssh-key@4.0.3:
        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
    - git-clone@4.0.12: {}
    - cache-pull@2.0.1: {}
    - script@1.1.5:
        inputs:
        - content: |2-

            # Sceengrab
            sudo gem install fastlane -NV
            fastlane init
            sudo gem install screengrab

            echo "
            locales ['en', 'th']
            clear_previous_screenshots true
            test_instrumentation_runner  'org.mozilla.focus.test.runner.ScreenshotTestRunner'
            use_tests_in_packages ['org.mozilla.focus.screengrab']
            app_package_name 'org.mozilla.rocket.debug.firebase'
            #use_tests_in_classes ['org.mozilla.focus.screengrab.MyShotScreenshot']
            tests_apk_path 'app/build/outputs/apk/androidTest/focusWebkit/firebase/app-focus-webkit-firebase-androidTest.apk'
            app_apk_path 'app/build/outputs/apk/focusWebkit/firebase/app-focus-webkit-firebase.apk'
            reinstall_app true
            exit_on_test_failure false
            " > ./fastlane/Screengrabfile
        title: Install Screengrab
    - install-missing-android-tools@2.3.3: {}
    - gradle-runner@1.8.3:
        inputs:
        - gradle_file: "$GRADLE_BUILD_FILE_PATH"
        - gradle_task: assembleFocusWebkitFirebase assembleFocusWebkitFirebaseAndroidTest
        - test_apk_file_exclude_filter: |-
            library-debug-androidTest.apk
            subsampling-scale-image-view-debug-androidTest.apk
        - apk_file_exclude_filter: |-
            *unaligned.apk
            *unsigned.apk
            *coverage.apk
            *Test*.apk
        - gradlew_path: "$GRADLEW_PATH"
    - script@1.1.5:
        inputs:
        - content: "#!/usr/bin/env bash\n# fail if any commands fails\nset -e\n# debug
            log\n#set -x\n\n$ANDROID_HOME/tools/bin/sdkmanager --install 'system-images;android-26;google_apis;x86'\n\n\ncd
            $ANDROID_HOME/emulator\n\necho no | avdmanager create avd -n Nexus_5X_API_26
            --device 'Nexus 5X' -k \"system-images;android-26;google_apis;x86\" --force\n\nemulator
            -avd Nexus_5X_API_26 -no-window -skin 1080x1920 -no-audio -debug-init
            > /dev/null 2>&1 &\n\necho \"Waiting emulator is ready...\"\n\nadb wait-for-device
            shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done;
            input keyevent 82'\n\necho \"Emulator is ready!\\n\"\n\nadb shell settings
            put global window_animation_scale 0 &\nadb shell settings put global transition_animation_scale
            0 &\nadb shell settings put global animator_duration_scale 0 &\n\n# we
            need this because we want reinstall_app set to true so that every time
            when fastlane screenshots runs, \n# it'll uninstall this new apk. if we
            don't add this, the first time fastlane screenshot runs will have error\nadb
            install $BITRISE_APK_PATH\nadb install $BITRISE_TEST_APK_PATH\n\necho
            \"animation disabled \\n\"\n\n\n"
        title: Start Enumlator
    - fastlane@2.3.12:
        inputs:
        - lane: screengrab
        is_always_run: true
    - script@1.1.5:
        inputs:
        - is_debug: 'yes'
        - content: "#!/usr/bin/env bash\ncp -r $BITRISE_SOURCE_DIR/app/build/reports/
            $BITRISE_DEPLOY_DIR/ \ncp -r ./fastlane/  $BITRISE_DEPLOY_DIR/ "
        is_always_run: true
        title: Copy build reports to deploy folder
    - deploy-to-bitrise-io@1.3.18:
        inputs:
        - is_compress: 'true'
        - notify_user_groups: none
```
And the result, is that there are some points still didn't work on Bitrise CI

```
[11:48:18]: ▸ [35mat org.mozilla.focus.screengrab.AaHomeScreenshot.screenshotHomeFragment(AaHomeScreenshot.java:82)[0m
        onView(withText(R.string.no_downloads)).check(matches(isDisplayed()));


[11:51:48]: ▸ [35mat org.mozilla.focus.screengrab.DownloadScreenshot.screenshotDownload(DownloadScreenshot.java:183)[0m
onView(allOf(withId(android.support.design.R.id.snackbar_text), withText(containsString(IMAGE_FILE_NAME_DOWNLOADED_PREFIX))))
                .check(matches(isDisplayed()));


[11:55:22]: ▸ [35mat org.mozilla.focus.screengrab.PrivateBrowsingScreenshot.screenshotPrivateBrowsing(PrivateBrowsingScreenshot.java:100)[0m
        onView(withId(R.id.pm_home_fake_input)).perform(click());


[12:04:01]: ▸ [35mat org.mozilla.focus.screengrab.OnboardingScreenshot.screenshotOnboardingPage(OnboardingScreenshot.java:92)[0m
// Dismiss adjust brightness dialog and open menu
        onView(withId(R.id.brightness_root)).perform(click());
        AndroidTestUtils.tapHomeMenuButton();
```

**Comment by chianger** [comment]
I believe it's done
